runtime/lua/vim/diagnostic.lua

```diff
@@ -228,6 +229,14 @@ end
 --- See |nvim_buf_set_extmark()|.
 --- @field virt_text_hide? boolean
 
+--- @class vim.diagnostic.Opts.VirtualLines
```

Comment:
`lsp_lines` also exposes an [option](https://git.sr.ht/~whynothugo/lsp_lines.nvim#usage) to highlight the entire diagnostic line. I didn't include that here, both as a matter of personal taste and because neither the pictures in the README nor Helix's usage have such highlighting. Open to discussion though.